metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -61,46 +61,51 @@ final class Embedded(
 }
 
 object Embedded {
-  def downloadSettings(
-      dependency: Dependency,
-      scalaVersion: String
-  ): Settings =
-    new coursiersmall.Settings()
-      .withTtl(Some(Duration.Inf))
-      .withDependencies(List(dependency))
-      .addRepositories(
-        List(
-          coursiersmall.Repository.SonatypeReleases,
-          coursiersmall.Repository.SonatypeSnapshots
+  lazy val repositories: List[Repository] =
+    Repository.defaults().asScala.toList ++
+      List(
+        Repository.central(),
+        Repository.ivy2Local(),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/releases/"
+        ),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/snapshots/"
         )
       )
-      .withForceVersions(
+
+  def fetchSettings(
+      dep: Dependency,
+      scalaVersion: String
+  ): Fetch = {
+
+    val resolutionParams = ResolutionParams
+      .create()
+      .forceVersions(
         List(
-          new Dependency(
-            "org.scala-lang",
-            "scala-library",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-compiler",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-reflect",
-            scalaVersion
-          )
-        )
+          Dependency.of("org.scala-lang", "scala-library", scalaVersion),
+          Dependency.of("org.scala-lang", "scala-compiler", scalaVersion),
+          Dependency.of("org.scala-lang", "scala-reflect", scalaVersion)
+        ).map(d => (d.getModule, d.getVersion)).toMap.asJava
       )
 
+    Fetch
+      .create()
+      .addRepositories(repositories: _*)
+      .withDependencies(dep)
+      .withResolutionParams(resolutionParams)
+      .withMainArtifacts()
+  }
+
   def newPresentationCompilerClassLoader(
       info: ScalaBuildTarget,
       scalac: ScalacOptionsItem
   ): URLClassLoader = {
-    val pc = new Dependency(
+    val scala_version = ScalaVersions
```

Comment:
nit: in Scala we use camel case instead of snake case
```suggestion
    val scalaVersion = ScalaVersions
```

Reply:
Fixed.